Send the email depending on the person who canceled the reservation.

andreaestefania12 opened the pull request:
## What does this PR do?

<!-- Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change. -->

Fixes #1708
_The event shows up in both user's upcoming bookings_: the solution for this first bug is to edit the filter when querying the bookings in apps/Web/server/routers/viewer.tsx, even though the booking is not showing in B's bookings, it blocks the availability of B.
 Although, in my opinion, it is better if you could reschedule or cancel the reservation, even if you are not the organizer.
<img width="280" alt="image" src="https://user-images.githubusercontent.com/19562383/162875807-81327387-c8b3-4488-b73e-cfc4c947535d.png">


## Type of change

<!-- Please delete options that are not relevant. -->

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

## How should this be tested?

<!-- Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration -->

For the first issue

1. Create 2 accounts, account A and B on cal.com.
2. Create an event in account A.
3. Schedule an event on account A using the email from account B.
4. Log in to account A and go to bookings/upcoming, you are going to see the booking.
5. Log in to account B and go to bookings/upcoming, you are not going to see the booking.

For the second issue

1. If an attendee cancels the reservation, the email will be sent to the organizer.

Commit:
Merge main into fix/cancel-emails-sends-to-wrong-account

Commit:
Merge conflicts

miguelnietoa commented on packages/types/Calendar.d.ts:
```diff
@@ -93,6 +93,7 @@ export interface CalendarEvent {
   cancellationReason?: string | null;
   rejectionReason?: string | null;
   hideCalendarNotes?: boolean;
+  isOrganizer: boolean;
```
Is the `isOrganizer` property required? 

Because this is causing a lot of errors when creating a `CalendarEvent`. Take a look into the [Check types workflow](https://github.com/kommitters/cal.com/runs/5991702217?check_suite_focus=true)

andreaestefania12 replied:
Thanks for making me realize that the `isOrganizer` property is not required, I'll change it to optional.

miguelnietoa commented on apps/web/pages/api/cancel.ts:
```diff
@@ -193,6 +197,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
       location: bookingToDelete.location ?? "",
       uid: bookingToDelete.uid ?? "",
       destinationCalendar: bookingToDelete?.destinationCalendar || bookingToDelete?.user.destinationCalendar,
+      isOrganizer: isOrganizer,
```
Same here!

miguelnietoa commented on apps/web/pages/api/cancel.ts:
```diff
@@ -127,6 +130,7 @@ export default async function handler(req: NextApiRequest, res: NextApiResponse)
     location: bookingToDelete?.location,
     destinationCalendar: bookingToDelete?.destinationCalendar || bookingToDelete?.user.destinationCalendar,
     cancellationReason: cancellationReason,
+    isOrganizer: isOrganizer,
```
When the property you want to send to the object is equal to the variable name, you could just send the variable like this:
```js
{
  isOrganizer,
}
```

miguelnietoa reviewed (changes requested):
In general, it looks good to me. 👌🏻 

But take a look into the checks because they are failing, and make the changes you need to.

I think maybe installing the extension Prettier together with ESLint can help you to detect the lint and type errors.

andreaestefania12 committed:
Merge branch 'calcom:main' into fix/cancel-emails-sends-to-wrong-account

andreaestefania12 committed:
Merge branch 'main' into fix/cancel-emails-sends-to-wrong-account

zomars committed:
Merge pull request #2672 from calcom/main

v1.5.4

zomars committed:
Merge branch 'main' into production

zomars committed:
Turbo fixes

zomars committed:
Make apps single pages public

hariombalhara committed:
Fix preview.html not built and thus served during depooy (#2713)

hariombalhara committed:
Fix time issue

hariombalhara committed:
Hotfix: Success page layout broken due to duplicate "When" (#2716)

PeerRich committed:
fixed /booking skeleton (#2722)

* fixed /booking skeleton

* nit

zomars committed:
Type fixes

zomars committed:
Test fixes

zomars committed:
Update playwright.config.ts

zomars committed:
More test fixes

zomars committed:
Update dynamic-booking-pages.test.ts

CarinaWolli committed:
add invite link to Zapier setup page (#2696)

* add invite link and toaster to zapier setup page

* create env variable for invite link and save in database

* fetch invite link form getStaticProps

* add getStaticPath method

* clean code

* Moves app setup and index page

* Moves Loader to ui

* Trying new way to handle dynamic app store pages

* Cleanup

* Update tailwind.config.js

* zapier invite link fixes

* Tests fixes

Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: zomars <zomars@me.com>

hariombalhara committed:
Add more embed events (#2719)

* Add more embed events

* Add more embed events

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

alishaz-polymath committed:
adds availability select loader (#2718)

hariombalhara committed:
Improve logs and Fix unwanted 500 to reduce noise in logs (#2674)

* Improve logs

* Fix unintentional 500

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

joeauyeung committed:
Change date format for RecurringBookings (#2707)

* Change date format for RecurringBookings

* Missing bookingId query param

Co-authored-by: Leo Giovanetti <hello@leog.me>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

PeerRich committed:
added giphy description (#2730)

PeerRich committed:
fixes #2732 (#2732)

hariombalhara committed:
Hotfix : Fix Infinite loading of Bookings (#2729)

* Add more embed events

* Add more embed events

* Fix nextCursor calculation logic

Co-authored-by: Peer Richelsen <peeroke@gmail.com>

leog committed:
Hotfix: Success page for recurring event (#2725)

* Merge pull request #2672 from calcom/main

v1.5.4

* Turbo fixes

* Make apps single pages public

* Fix preview.html not built and thus served during depooy (#2713)

* Hotfix: Success page layout broken due to duplicate "When" (#2716)

* Update BookingPage.tsx

* Reverting unchaged lines

* Fixing recurrenceRule for ICS files

Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: Hariom Balhara <hariombalhara@gmail.com>

leog committed:
Hotfix: Success page for recurring event (#2725)

* Merge pull request #2672 from calcom/main

v1.5.4

* Turbo fixes

* Make apps single pages public

* Fix preview.html not built and thus served during depooy (#2713)

* Hotfix: Success page layout broken due to duplicate "When" (#2716)

* Update BookingPage.tsx

* Reverting unchaged lines

* Fixing recurrenceRule for ICS files

Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: Hariom Balhara <hariombalhara@gmail.com>

leog committed:
Missing fix for success page

hariombalhara committed:
Fix preview.html not built and thus served during depooy (#2727)

Co-authored-by: Omar López <zomars@me.com>

alishaz-polymath committed:
Allow deletion of a disabled event (#2737)

* allows deletion of disabled event

* some visual fixes

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

zomars committed:
Multiple E2E improvements

zomars committed:
Parallelizes some tests

zomars committed:
Update booking-pages.test.ts

zomars committed:
E2E and paid bookings fixes

emrysal committed:
Add 'free' and 'workingElsewhere' as a non-blocking event (#2652)

* Add 'free' and 'workingElsewhere' to non-blocking event - this will allow bookings at these times

* Update CalendarService.ts

Co-authored-by: zomars <zomars@me.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

zomars committed:
TODO marks blocking test to be fixed

zomars committed:
Update testUtils.ts

zomars committed:
Update testUtils.ts

zomars committed:
getBusyTimes consolidation

zomars committed:
Fixes delete-me test

zomars committed:
E2E fixing attemps

CarinaWolli committed:
Adjusting Zapier endpoints for publishing integration (#2728)

* add /me endpoint for zapier API testing

* remove cacellationReason from listBookings response

Co-authored-by: CarinaWolli <wollencarina@gmail.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: Omar López <zomars@me.com>

hariombalhara committed:
Fix reschedule not happening in calendar if two calendards are there (#2733)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>

zomars committed:
Increases timeout temporarily

joeauyeung committed:
Merge pull request #2745 from calcom/apps/multiple-categories

Allow apps to belong to multiple categories

zomars committed:
Build fixes

joeauyeung committed:
Populate msteams key in db (#2743)

* Populate msteams key in db

* Fix calendar credentials to teams

* Clarify account dialog

Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: Bailey Pumfleet <pumfleet@hey.com>

zomars committed:
Improving Email DRYness (#2486)

* Email DRY

* WIP

* Improve email DRYness

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

zomars committed:
E2E fixtures (#2747)

gitstart committed:
Sign in button should be changed or disabled after click #2654 (#2749)

Co-authored-by: gitstart <gitstart@users.noreply.github.com>

leog committed:
Fixing ESLint warnings (#2723)

* Fixing warnings

* Reverting and disabling ESLint in some cases

* Reverting Next Images

* Reverting file, bad merge

* Targeting ESLint to line

* Additional warnings

* New warning squished

* More tweaks and major fixes

* Uneeded conf

Co-authored-by: Omar López <zomars@me.com>

arthur1041 committed:
Fix/avoid multiple schedule deletions (#2602)

* Prevents users from deleting the same schedule multiple times due to delay before the schedule disappears. It also applies the same fix to team disband.

Schedule deletion:
![schedule_deletion_new_behaving](https://user-images.githubusercontent.com/42497300/165126805-b3090268-c1a6-418a-b06e-06bd8446da03.gif)

Team disband:
![team_disband_new_behaving](https://user-images.githubusercontent.com/42497300/165127043-7e083e94-e4c9-4e88-90a2-47d31bdd92e6.gif)

Fixes issue [#2569](https://github.com/calcom/cal.com/issues/2569)

Bug fix (non-breaking change which fixes an issue)

**apps/web/components/LightLoader.tsx** → this file was created in order to make a light color loading spinner available. It's necessary when we need to display a loading spinner above dark backgrounds.

**apps/web/components/availability/ScheduleListItem.tsx** → this component was created in order to give a schedule list item its own state.

* Removing a "setTimeout" that was only used for testing purposes

* Adding a code review suggestion to my modifications

* Changing loading style

* Cleanup

* Avoids using unnecessary state

* Revert "Adding a code review suggestion to my modifications"

This reverts commit b5e40062d71157ca1d9384fb1f5c30d50625809d.

* Reverts some changes

* Renames isLoading

Co-authored-by: Alex van Andel <me@alexvanandel.com>
Co-authored-by: Omar López <zomars@me.com>

Mitchell-Moore committed:
Added check on create eventtype to see is user has slug that already exists (#2757)

* Added check on create eventtype to see is user has slug that already exists, added error check on onError

* revert yarn.lock back

* Catches prisma known error instead of making an additional query

Co-authored-by: zomars <zomars@me.com>

hariombalhara committed:
Fix white border (#2761)

Commit:
fix: remove hardcoded redirect in signin url email verification (#2764)

Co-authored-by: Agusti Fernandez Pardo <git@agusti.me>

Commit:
fix: remove hardcoded redirect in signin url email verification (#2764)

Co-authored-by: Agusti Fernandez Pardo <git@agusti.me>

gitstart committed:
Bad UX when user wants to set the default Event Type Title #2245 (#2760)

Co-authored-by: gitstart <gitstart@users.noreply.github.com>

zomars committed:
Skip sending emails in E2E

sean-brydon committed:
Users Phone Number Option (#2669)

* Users Phone Number Option

* Implemented improvments

* Add validation to form

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: zomars <zomars@me.com>

andreaestefania12 committed:
Fix/unpaid unconfirmed (#2553)

* Fix merge errors

* Errors prettier/prettier

* Update apps/web/pages/api/book/event.ts

Co-authored-by: Miguel Nieto A <39246879+miguelnietoa@users.noreply.github.com>

* Update apps/web/pages/api/book/event.ts

Co-authored-by: Miguel Nieto A <39246879+miguelnietoa@users.noreply.github.com>

* Update apps/web/pages/api/integrations.ts

Co-authored-by: Miguel Nieto A <39246879+miguelnietoa@users.noreply.github.com>

* Fix merge errors

* Errors prettier/prettier

* Update apps/web/pages/api/book/confirm.ts

Co-authored-by: alannnc <alannnc@gmail.com>

* Modal window before delete stripe integration

* ESLint Report

* Test fixes

Co-authored-by: Miguel Nieto A <39246879+miguelnietoa@users.noreply.github.com>
Co-authored-by: alannnc <alannnc@gmail.com>
Co-authored-by: zomars <zomars@me.com>

leog committed:
Mutually exclusive options (#2755)

Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: Peer Richelsen <peeroke@gmail.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

gitstart committed:
Readd steps to create a new user #2665 (#2759)

Co-authored-by: gitstart <gitstart@users.noreply.github.com>

Hashen110 committed:
remove redundant conditional expressions (#2756)

* remove redundant conditional expressions

* remove redundant conditional expression

Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>

zomars committed:
Merge branch 'main' into pr/2552

zomars committed:
Update organizer-cancelled-email.ts

zomars committed:
Reverting diffs

Commit:
Fix adds redirect callback that support app.cal.com (#2768)

* Fix adds redirect callback that support app.cal.com

* Update apps/web/pages/api/auth/[...nextauth].tsx

Check origin of website and baseurl

Co-authored-by: Omar López <zomars@me.com>

* fix: lint issue extra space removed

Co-authored-by: Agusti Fernandez Pardo <git@agusti.me>
Co-authored-by: Omar López <zomars@me.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

buschco committed:
respect local set timezone and update url on mismatch (#2506)

* ensure `timeZone()` will make its way to the URL

fixes https://github.com/calcom/cal.com/issues/2482

* keep `timeZone()` and the offset from URL in sync

Co-authored-by: Syed Ali Shahbaz <52925846+alishaz-polymath@users.noreply.github.com>
Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

zomars committed:
Squashed commit of the following:

commit 27540b09ce950f215475ddffe15dd8f9b3877f5b
Author: Agusti Fernandez Pardo <me@agusti.me>
Date:   Mon May 16 17:34:13 2022 +0200

    fix: remove hardcoded redirect in signin url email verification (#2764)

    Co-authored-by: Agusti Fernandez Pardo <git@agusti.me>

commit ae15a7d7398258a0bff23d24d9a067750e875bd7
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Tue May 10 14:30:43 2022 +0530

    Fix time issue

commit 2a5a89fe50bf181cbbe4124eb8ed967545bd377a
Author: Leo Giovanetti <hello@leog.me>
Date:   Wed May 11 10:21:46 2022 -0300

    Missing fix for success page

commit 2ce1e780537f48f81a674d752b1301f6cf6adc23
Author: Leo Giovanetti <hello@leog.me>
Date:   Wed May 11 10:12:59 2022 -0300

    Hotfix: Success page for recurring event (#2725)

    * Merge pull request #2672 from calcom/main

    v1.5.4

    * Turbo fixes

    * Make apps single pages public

    * Fix preview.html not built and thus served during depooy (#2713)

    * Hotfix: Success page layout broken due to duplicate "When" (#2716)

    * Update BookingPage.tsx

    * Reverting unchaged lines

    * Fixing recurrenceRule for ICS files

    Co-authored-by: Omar López <zomars@me.com>
    Co-authored-by: Hariom Balhara <hariombalhara@gmail.com>

commit 2d6d1cb444e3a95ae85e50235a76d716156a41dd
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Tue May 10 14:49:46 2022 +0530

    Hotfix: Success page layout broken due to duplicate "When" (#2716)

commit ef68f4f4f8f2195de33af15946eec52ab4818b40
Author: Hariom Balhara <hariombalhara@gmail.com>
Date:   Tue May 10 10:54:20 2022 +0530

    Fix preview.html not built and thus served during depooy (#2713)

commit 18c28cc3fdae3e4f4396ecb974c8593736be7269
Author: zomars <zomars@me.com>
Date:   Mon May 9 16:17:07 2022 -0600

    Make apps single pages public

commit d40e8caff9d201c5a6a99ac49ca0a870798ff48c
Author: zomars <zomars@me.com>
Date:   Mon May 9 16:08:03 2022 -0600

    Turbo fixes

commit 3161cc4d4520a7a214349605fa2bae4184cd5931
Merge: ed808c3be 4099a477d
Author: zomars <zomars@me.com>
Date:   Mon May 9 14:58:33 2022 -0600

    Merge branch 'main' into production

commit ed808c3be61e3858f559489e069dd87d35bf2e12
Author: Omar López <zomars@me.com>
Date:   Mon May 9 14:56:23 2022 -0600

    Merge pull request #2672 from calcom/main

    v1.5.4

zomars committed:
Merge branch 'production' into main

zomars committed:
Typo

zomars committed:
Typo

zomars committed:
Update apps/web/pages/apps/categories/[category].tsx

zomars committed:
Apply suggestions from code review

leog committed:
Alert to describe exclusion of options (#2770)

* Alert to describe exclusion of options

* Update apps/web/pages/event-types/[type].tsx

Co-authored-by: Omar López <zomars@me.com>

* Update apps/web/components/eventtype/RecurringEventController.tsx

Co-authored-by: Omar López <zomars@me.com>

* Formatting

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: Omar López <zomars@me.com>

iamkun committed:
fix: split time correctly if the local working hours are just across mid night (#2766)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

zomars committed:
Update crowdin.yml

github-actions[bot] committed:
New Crowdin translations by Github Action (#2773)

* New Crowdin translations by Github Action

* Update vital.json

* Update vital.json

Co-authored-by: Crowdin Bot <support+bot@crowdin.com>
Co-authored-by: Omar López <zomars@me.com>

joeauyeung committed:
Add Google cal extneral calendar id to booking reference (#2671)

* Set google cal event id to use our uid

* Save calendar external id to bookingRef

* Pass external calendar ids to update and delete

* Create migration

* Fix type errors

* Fix prisma url

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>
Co-authored-by: Omar López <zomars@me.com>

zomars committed:
v1.6

zomars committed:
v1.6

andreaestefania12 committed:
Update apps/web/pages/api/cancel.ts

Co-authored-by: Omar López <zomars@me.com>

andreaestefania12 committed:
Merge branch 'main' into fix/cancel-emails-sends-to-wrong-account

andreaestefania12 committed:
Update Calendar.d.ts

andreaestefania12 committed:
Update cancel.ts

sean-brydon committed:
2FA submit disabled (#2790)

agordhandas committed:
fixing the hyperlink for open startup (#2777)

Co-authored-by: kodiakhq[bot] <49736102+kodiakhq[bot]@users.noreply.github.com>

hariombalhara committed:
Add login event (#2784)

Co-authored-by: Bailey Pumfleet <pumfleet@hey.com>

Commit:
Merge branch 'main' into fix/cancel-emails-sends-to-wrong-account
